src/index.vue

```diff
@@ -1,42 +1,58 @@
 <script>
+
+const supportedModifiers = ['altKey', 'metaKey', 'ctrlKey', 'shiftKey']
+
 export default {
   props: {
-    keyCode: {
-      type: Number,
-      default: null
-    },
-    modifiers: {
-      type: Array, // ['shiftKey', 'ctrlKey', 'altKey', 'metaKey']
-      default: () => []
-    },
-    event: {
-      type: String,
-      default: "keyup"
-    },
-    preventDefault: {
-      type: Boolean
-    }
+    config: {
+      type: Array,
+      default: [{
+        keyCodes: null,
+        events: ['keyup'],
+        modifiers: null
+      }]
+    }    
   },
   mounted() {
-    window.addEventListener(this.event, this.emitEvent);
+    this._keyListeners = []    
```

Comment:
Please don't use _ for variables in vue. Doesn't do much difference here in this case, but I wanna avoid it anyway since it's bad practice.
See [documentation](https://vuejs.org/v2/api/#data).

Reply:
The 'bad practice' you refer to only applies to reactive items in the 'data' field, not internal component properties that should not be accessed externally. So would you rather have no idoimatic difference between non-reactive and reactive properties? The keyListeners are not meant to be accessed in anyway except in the implementation.